### PR TITLE
Add struct and functionality for using user input 

### PR DESCRIPTION
**Pull request by rbehjati**
Ref #1191

Builds on top of #1386 and adds struct and functionality for parsing and using user-provided input. 

Moves `InputOptions` to package `pkg` to be able to pass it as an argument to `NewDockerBuildConfig` without introducing a circular dependency.

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]

**Review by ianlewis — APPROVED** [review]

File: internal/builders/docker/pkg/config_test.go
Comment (rbehjati) [reply internal/builders/docker/pkg/config_test.go]: Thanks. Fixed.

File: internal/builders/docker/testdata/config.toml
Comment (rbehjati) [reply internal/builders/docker/testdata/config.toml]: Done.

File: internal/builders/docker/pkg/config.go
Comment (rbehjati) [reply internal/builders/docker/pkg/config.go:44]: Done.

File: internal/builders/docker/pkg/config.go
Comment (rbehjati) [reply internal/builders/docker/pkg/config.go:166]: Thanks. Replaced the validation check above that only checked that the path is relative with a call to `utils.PathIsUnderCurrentDirectory`.

**Review by rbehjati — COMMENTED** [review]
Thanks for the review.

File: internal/builders/docker/commands.go
Comment (rbehjati) [reply internal/builders/docker/commands.go]: Done.

File: internal/builders/docker/pkg/config.go
Comment (rbehjati) [inline internal/builders/docker/pkg/config.go:72]: I could make the fields in `DockerBuildConfig` private/unexposed to make sure it is not possible to create an instance of it without performing these checks. What do you think? @asraa & @ianlewis (somewhat related to the comment below on validating the path)

**Comment by rbehjati** [comment]
@asraa @laurentsimon @ianlewis I am happy with this PR. Can you please merge it?

**Comment by rbehjati** [comment]
> Can you rebase and solve the conflicts?

Done. But now the CI step for `verify base image` fails. I don't think that is due to my changes. Is it?

**Comment by laurentsimon** [comment]
No it's not your code. @ianlewis thoughts on what's going on?
I'm not sure there is a commitment from the image maintainers to sign the images. I'll merge this PR to unblock you.
Filed https://github.com/slsa-framework/slsa-github-generator/issues/1401 to resolve later.

**Comment by rbehjati** [comment]
Many thanks :)

**Pull request merged**
